src/relay/src/hooks/useRelayEnvironment.js

```diff
@@ -1,9 +1,16 @@
 // @flow
 
-import { useRelayEnvironment as _useRelayEnvironment } from 'react-relay/hooks';
+import * as React from 'react';
+import { ReactRelayContext } from 'react-relay';
+import { invariant } from '@adeira/js';
 
 import type { Environment } from '../runtimeTypes.flow';
 
 export default function useRelayEnvironment(): Environment {
-  return _useRelayEnvironment();
+  const context = React.useContext(ReactRelayContext);
+  invariant(
+    context != null,
+    'Expected to be used inside a relay context, but was not. Make sure it is wrapped in a RelayEnvironmentProvider or QueryRenderer',
+  );
+  return context.environment;
```

Comment:
This is what we used before (basically just a better error message):

```js
// @flow

import { useContext } from 'react';
import { ReactRelayContext } from 'react-relay';
import { invariant } from '@adeira/js';

import type { Environment } from '../runtimeTypes.flow';

/**
 * You can use this hook in any component below QueryRenderer or RelayEnvironmentProvider.
 */
export default function useRelayEnvironment(): Environment {
  const context = useContext(ReactRelayContext);
  invariant(
    context != null,
    'useRelayEnvironment: Expected to have found a Relay environment provided by' +
      'a `RelayEnvironmentProvider` component or `QueryRenderer`. This usually means that' +
      'useRelayEnvironment was used in a component that is not a descendant of these components. ' +
      'Please make sure a `RelayEnvironmentProvider` has been rendered somewhere as a parent of ' +
      'your component. It should be defined only once in the whole application.',
  );
  return context.environment;
}
```

Reply:
Sure, I will use the one from before

Reply:
@mrtnzlml please recheck with new error message 🚀